docs/source/basic_features.rst

```diff
@@ -0,0 +1,249 @@
+
+The command queue
+-----------------
+Control flow commands, register access and memory access are all done through the command queue. This is a FIFO queue of commands that are executed in order. 
+
+While the inner workings of the command queue are transparent to the user, it is important to understand how its handling impacts the flow of the debugging script:
+
+By default, a command is polled from the queue when the execution stops, either as a result of handling a breakpoint, a signal, or other similar events.
+In the following example, the content of the RAX register is printed after the program hits the breakpoint or stops for any other reason:
+
+.. code-block:: python
+
+    d = debugger("program")
+
+    d.breakpoint("func")
+
+    d.cont()
+
+    print(f"RAX: {hex(d.regs.rax)}")
+
+This flow is similar to how a GDB script would work, allowing for a more intuitive starting point. If you would like to have more control, however, you can disable this behavior to make sure the command queue is polled as soon as possible.
+
+This can be done by setting the `auto_interrupt_on_command` parameter to True when creating the debugger object. In this new scenario, we would have to modify the script to recreate the previous flow.
+
+.. code-block:: python
+
+    d = debugger("program", auto_interrupt_on_command=True)
+
+    d.breakpoint("func")
+
+    d.cont()
+    d.wait()
+
+    print(f"RAX: {hex(d.regs.rax)}")
+
+The `wait()` method waits for the running process to stop before going forward with the script. Adding the `d.wait()` command will make sure the register access doesn't happen before hitting the breakpoint or any other stopping event. If the `wait()` method is omitted, the register access will happen as soon as possible after the continue command is issued. Please remember that accessing a property like registers will stop the process. Sending a continue command afterwards will make the process run again.
+
+
+You can manually send a stopping signal to the program using the `interrupt()` method. This will stop the execution of the program and allow you to access the registers and memory. The syntax is as follows:
+
+.. code-block:: python
+
+    d.interrupt()
+
+Register Access
+===============
+.. _register-access-paragraph:
+
+libdebug offers a simple register access interface for supported architectures. The registers are accessed through the regs attribute of the debugger object. The field includes both general purpose and special registers, as well as the flags register. Effectively, any register that can be accessed by an assembly instruction, can also be accessed through the regs attribute.
```

Comment:
it is an attribute of the ThreadContext actually. The debugger exposes only the one of the main thread. I would clarify this.